blog/2025-06-15-temporal-impl-1.md

```diff
@@ -0,0 +1,439 @@
+overflow option to constrain or reject based on whether the provided
+values are in an expected range.
+
+However, we can better express this in Rust with common `try_` prefix
+notation.
+
+```rust
+impl PlainDate {
+    pub fn new(year: i32, month: u8, day: u8, calendar: Calendar) -> Result<Self> {
+        Self::new_with_overflow(year, month, day, calendar, ArithmeticOverflow::Constrain)
+    }
+
+    pub fn try_new(year: i32, month: u8, day: u8, calendar: Calendar) -> Result<Self> {
+        Self::new_with_overflow(year, month, day, calendar, ArithmeticOverflow::Reject)
+    }
+}
+```
+
+These three constructors, `new_with_overflow`, `try_new`, and `new`, are
+fairly flexible and provide full coverage of the Temporal specification.
+
+For instance, take the below constructor:
+
+```js
+const plainDate = new Temporal.PlainDate(2025, 6, 9);
+```
+
+Can easily be conveyed in Rust as:
+
+```rust
+use temporal_rs::PlainDate;
+let plain_date = PlainDate::try_new(2025, 6, 9, Calendar::default())?;
+```
+
+We actually learn some interesting things immediately about the
+JavaScript API from looking at the `temporal_rs` API:
+
+1. `Temporal.PlainDate` constructor can throw.
+2. When the calendar is omitted, the default calendar is used (this will
+   default to the `iso8601` calendar)
+
+Of course, if you somewhat prefer the brevity of the JavaScript API and
+don't want to list the default `Calendar`, `temporal_rs` provides the
+additional constructors `new_iso` and `try_new_iso`.
+
+```rust
+use temporal_rs::PlainDate;
+let plain_date = PlainDate::try_new_iso(2025, 6, 9)?;
+```
+
+Interestingly enough, the `_iso` constructors are actually extensions of
+Temporal specification to provide a similar API in Rust. This is because
+the `_iso` constructors are assumed to exist due to resolving an
+`undefined` calendar to the default ISO calendar.
+
+## Let's discuss `Now`
+
+> Colonel Sandurz: Now. You're looking at now, sir. Everything that
+> happens now, is happening now.<br></br> Dark Helmet: What happened to
+> then?<br></br> Colonel Sandurz: We passed then.<br></br> Dark Helmet:
+> When?<br></br> Colonel Sandurz: Just now. We're at now now.<br></br>
+> Dark Helmet: Go back to then.<br></br> Colonel Sandurz: When?<br></br>
+> Dark Helmet: Now.<br></br> Colonel Sandurz: Now?<br></br> Dark Helmet:
+> Now.<br></br> Colonel Sandurz: I can't.<br></br> Dark Helmet:
+> Why?<br></br> Colonel Sandurz: We missed it.<br></br> Dark Helmet:
+> When?<br></br> Colonel Sandurz: Just now.<br></br> Dark Helmet: When
+> will then be now?<br></br> -- <cite>Spaceballs, 1987</cite>
+
+`Temporal.Now` is an incredibly strange type, yet nevertheless
+important. It is the object from which the current instant can be
+measured and mapped into any of the Temporal components.
+
+In JavaScript, this type has no `[[Construct]]` or `[[Call]]` internal
+method, which is a fancy way to say that Now has no constructor and
```

Comment:
```suggestion
In JavaScript, this type has no `[[Construct]]` or `[[Call]]` internal
methods, which is a fancy way to say that Now has no constructor and
```

Also, we should link to https://tc39.es/ecma262/multipage/ordinary-and-exotic-objects-behaviours.html#sec-ecmascript-function-objects-call-thisargument-argumentslist and https://tc39.es/ecma262/multipage/ordinary-and-exotic-objects-behaviours.html#sec-ecmascript-function-objects-construct-argumentslist-newtarget for the curious ones.